x/ibc/core/04-channel/keeper/handshake.go

```diff
@@ -104,68 +100,71 @@ func (k Keeper) ChanOpenTry(
 	order types.Order,
 	connectionHops []string,
 	portID,
-	desiredChannelID,
-	counterpartyChosenChannelID string,
+	previousChannelID string,
 	portCap *capabilitytypes.Capability,
 	counterparty types.Counterparty,
 	version,
 	counterpartyVersion string,
 	proofInit []byte,
 	proofHeight exported.Height,
-) (*capabilitytypes.Capability, error) {
-	// channel identifier and connection hop length checked on msg.ValidateBasic()
-	previousChannel, found := k.GetChannel(ctx, portID, desiredChannelID)
-	if found && !(previousChannel.State == types.INIT &&
-		previousChannel.Ordering == order &&
-		previousChannel.Counterparty.PortId == counterparty.PortId &&
-		previousChannel.Counterparty.ChannelId == counterparty.ChannelId &&
-		previousChannel.ConnectionHops[0] == connectionHops[0] &&
-		previousChannel.Version == version) {
-		return nil, sdkerrors.Wrap(types.ErrInvalidChannel, "cannot relay connection attempt")
+) (string, *capabilitytypes.Capability, error) {
+	channelID := previousChannelID
+
+	// empty channel identifier indicates continuing a previous channel handshake
+	if previousChannelID != "" {
+		// channel identifier and connection hop length checked on msg.ValidateBasic()
+		// ensure that the previous channel exists
+		previousChannel, found := k.GetChannel(ctx, portID, previousChannelID)
```

Comment:
nit: you can refactor to call this once so we don't have to requery store when we create capability on line 199

Reply:
I didn't do this initially because getting the connection also uses `found` so I added a different var `previousChannelFound`. I think the naming also helps given the distance between the two uses is the top and bottom of the function